lib/3scale/backend/service.rb

```diff
@@ -176,8 +176,15 @@ def default_service?(provider_key, id)
       end
 
       def initialize(attributes = {})
-        # default state
-        @state = :active
+        # :state is set as active in this method when:
+        # - The state key is not present in the attributes hash
+        # - The state key is present in the attributes hash but it has
+        #   the nil value
+        # This is done in order to not break compatibility for existing
+        # Services saved in the database, that do not contain the state
+        # key.
+        attributes[:state] ||= :active
```

Comment:
this change is not required
```
@state = :active
```
is enough

Reply:
I agree that it's enough with @state = :active but I think it's better for the readability of the code to leave the attributes[:state] ||= :active  operation because it shows the clear intent.

Reply:
You are checking if value is nil in attribute writer accessor, no need to double check here.

Reply:
I know, it's just a matter of readability of the code.

Reply:
this is not coding style. This is different logic implementation. Currently same result for the current implementation of the attribute writer.

Reply:
With both options the same result would be returned. I prefer to maintain the readability at this moment